packages/nuxt/src/app/plugins/http-client-hints/http-client-hints.server.ts

```diff
@@ -0,0 +1,292 @@
+import type { IncomingHttpHeaders, ServerResponse } from 'node:http'
+import { readonly } from 'vue'
+import type { HttpClientHintsConfiguration, HttpClientHintsRequest } from './types'
+import { type Browser, parseUserAgent } from './detect-browser'
+import { NuxtHTTPClientHintsState } from './types'
+import { defineNuxtPlugin, useNuxtApp, useRuntimeConfig } from '#app/nuxt'
+import { useCookie } from '#app/composables/cookie'
+import { useState } from '#app/composables/state'
+import { useRequestEvent } from '#app/composables/ssr'
+
+export default defineNuxtPlugin<{ httpClientHints: HttpClientHintsRequest }>({
+  name: 'nuxt:http-client-hints:server',
+  enforce: 'pre',
+  setup () {
+    const event = useRequestEvent()
+    const httpClientHintsConfiguration = useRuntimeConfig().public.httpClientHints as HttpClientHintsConfiguration
+    const httpClientHints = useState<HttpClientHintsRequest>(NuxtHTTPClientHintsState)
+
+    const request = event.node.req
+    const response = event.node.res
+
+    const requestHeaders = request.headers ?? {}
+
+    const userAgentHeader = readClientHeader('user-agent', requestHeaders)
+
+    // 1. extract browser info
+    const userAgent = userAgentHeader
+      ? parseUserAgent(userAgentHeader)
+      : null
+    // 2. prepare client hints request
+    const clientHintsRequest = collectClientHints(userAgent, httpClientHintsConfiguration, requestHeaders)
+    // 3. write client hints response headers
+    writeClientHintsResponseHeaders(clientHintsRequest, httpClientHintsConfiguration, response)
```

Comment:
Same here, we should try to avoid nodejs ServerResponse

Reply:
done